scripts/build/setup/termux_setup_nim.sh

```diff
@@ -0,0 +1,84 @@
+termux_setup_nim() {
+	local CHOOSENIM_VERSION=0.8.16
+	local CHOOSENIM_URL="https://github.com/nim-lang/choosenim/releases/download/v${CHOOSENIM_VERSION}/choosenim-${CHOOSENIM_VERSION}_linux_amd64"
+	local CHOOSENIM_FILE=${TERMUX_PKG_TMPDIR}/choosenim
+	local CHOOSENIM_SHA256="1034e03564b3e5b5321646f9ce290cc4d3cdff0cf0e9b523d43f0db8b9a45c74"
+
+	local NIM_FOLDER
+	local NIM_PKG_VERSION="$(
+		. "$TERMUX_SCRIPTDIR/packages/nim/build.sh"
+		echo "$TERMUX_PKG_VERSION"
+	)"
+
+	if [[ -z "${NIM_VERSION-}" ]]; then
+		NIM_VERSION=${NIM_PKG_VERSION}
+	fi
+
+	if [ "${TERMUX_PACKAGES_OFFLINE-false}" = "true" ]; then
+		NIM_FOLDER=${TERMUX_SCRIPTDIR}/build-tools/nim-${NIM_VERSION}
+		CHOOSENIM_FOLDER=${TERMUX_SCRIPTDIR}/build-tools/choosenim-${CHOOSENIM_VERSION}
+	else
+		NIM_FOLDER=${TERMUX_COMMON_CACHEDIR}/nim-$NIM_VERSION
+		CHOOSENIM_FOLDER=${TERMUX_COMMON_CACHEDIR}/choosenim-$CHOOSENIM_VERSION
```

Comment:
Declare NIM_FOLDER and CHOOSENIM_FOLDER before "if" so you can remove the "else"